### ISSUE TITLE
Expose method of subscribing to form data

### ISSUE THREAD

**Issue by philholden** [final-form/react-final-form#41]
Feature request: I would like to be able to subscribe to form updates for the purpose of autosave without submission. I am currently using `FormSpy` to do this but its render function is also triggered by parents rerendering:

```javascript
<FormSpy subscription={{ values: true }}>
   {({ values }) => {
     onChange(values);
     return null;
   }}
</FormSpy>
```
It would nice if there was easy way to listen to updates without it also trigging on renders and gives the granularity subscriptions offer (`debug` is too noisy). I could get hold of `reactFinalForm` from `context` but this in not part of the public API.

One of the following would work for me:

* An `onChange` prop on `FormSpy` 
* A simular component that accepts `subcription` and `onChange` as `props`
* A component that fetches `reactFinalForm` from the `context` 
* `reactFinalForm` being exposed as a render prop on Form

_Am I missing something, is there already a way to do this?_

**Comment by erikras** [comment]
Wow, your analysis of the situation is flawless! 👍

> - An onChange prop on FormSpy

☝️ I think this is the best option. And if you provide an `onChange`, then you don't need to provide any rendering information.

Another option you missed would be to put an `onChange` prop on `Form` itself – that would default to only values ([`redux-form` has this](https://redux-form.com/7.2.0/docs/api/reduxform.md/#-code-onchange-function-code-optional-)) – but it feels better on `FormSpy`, since you can alter the subscription to be whatever part of state you want.

**Comment by philholden** [comment]
Yes `onChange` on the `Form` makes a lot of sense as `Field` already has `onChange`. I did try `onChange` intuitively as a `prop` before reading the docs. So it may be nice to add this for common use cases.

`FormSpy` would be nice to give more control over `subscription`s. And you can have lots of spys with different `subscriptions`.

**Comment by erikras** [comment]
> Yes `onChange` on the Form makes a lot of sense as Field already has `onChange`. 

`Field` does not have an `onChange` prop. It does in Redux Form, but it does not (yet) in React Final Form.

**Comment by erikras** [comment]
No, I want to solve this another way.

**Comment by philholden** [comment]
Thanks

**Comment by erikras** [comment]
Published fix in [`v1.2.0`](https://github.com/final-form/react-final-form/releases/tag/v1.2.0).
